src/main/java/org/cactoos/iterable/Matched.java

```diff
@@ -92,7 +92,7 @@ public Matched(
                         );
                     }
                 }
-                return Collections.unmodifiableList(rslt).iterator();
+                return new Immutable<>(rslt).iterator();
```

Comment:
@andreoss could you also, in `list.Immutable.iterator()`, wrap the iterator of the wrapped list in a `iterator.Immutable`.

Reply:
@victornoel Done

Reply:
@andreoss I think you misunderstood my comment. The code you had was ok, I meant to change the implementation in `list.Immutable` because there it is incorrect.

Reply:
@victornoel Sorry, corrected that.